src/wasm/wasm-type.cpp

```diff
@@ -1211,7 +1272,8 @@ struct TypeBuilder::Impl {
       info = std::make_unique<HeapTypeInfo>(Signature());
     }
     void set(HeapTypeInfo&& hti) {
-      *info = hti;
+      *info = std::move(hti);
```

Comment:
is this just for clarity?

Reply:
C++ is weird because even if you have an rvalue (&&) reference, you still need to use std::move in order to get move semantics, so this code was not working as intended before. (Rust does not have such pitfalls 😉) See https://stackoverflow.com/questions/31213539/c-stdmove-with-rvalue-reference-is-not-moving-contents.

Reply:
Ah, I did not know that... well, `std::move` everywhere then... :shrug: